src/include/ibase.h

```diff
@@ -64,6 +64,14 @@
 #define FB_API_DEPRECATED
 #endif
 
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32__)
+#define FB_DLL_EXPORT __declspec(dllexport)
+#elif defined(__APPLE__) || defined(__linux__) || defined(unix) || defined(__unix__) || defined(__unix)
+#define FB_DLL_EXPORT __attribute__ ((visibility("default")))
```

Comment:
I'm not sure - does \_\_attribute\_\_ ((visibility("default"))) work in non-GCC case? I.e. is not check for unix too wide, may be something like check for defined(\_\_GNUC\_\_) should be added?

Reply:
Another problem here is that __declspec() must be used BEFORE function name while __attribute__() - AFTER it. I.e. declaration `void FB_DLL_EXPORT foo()` won't work with GCC.

Reply:
> I'm not sure - does __attribute__ ((visibility("default"))) work in non-GCC case? I.e. is not check for unix too wide, may be something like check for defined(__GNUC__) should be added?

Or maybe this?

https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005fattribute.html